### PR TITLE
Package init

### PR DESCRIPTION
**Pull request by ibilous**

### CODE REVIEW

File: analyticConfigurations/ace-efs.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "analyticConfiguration": {
+    "metrics": [
+      {
+        "match": "aws.efs.burstcreditbalance.*",
```
Comment (TheConnMan) [inline analyticConfigurations/ace-efs.json]: Do you need the `.*` at the end? `aws.efs.burstcreditbalance` is the full metric FQN.

File: analyticConfigurations/metric_meta-efs.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{
+  "analyticConfiguration": {
+    "metrics": [
+      {
+        "match": "aws.efs.burstcreditbalance.*",
+        "properties": {
+          "statistic": "MAX",
```
Comment (TheConnMan) [inline analyticConfigurations/metric_meta-efs.json]: I don't think these are counters. Take a look at https://docs.aws.amazon.com/efs/latest/ug/monitoring-cloudwatch.html for which statistics should be used. I believe the byte metrics should be SUM and the rest should be AVERAGE (the default). I also think all the types should be defaults.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
+{
+  "id": "netuitive.packages.aws.efs",
+  "version": "1.0.0",
+  "manifest": {
+    "name": "Netuitive Packages Elastic File System",
+    "description": "A set of Netuitive analytics configurations, polices, dashboards, and reports that are used to monitor performance of the elements collected from the ElasticSearch collector in the Netuitive Agent.",
```
Comment (TheConnMan) [inline package.json]: This description needs updating.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    "author": {
+      "name": "Netuitive Research",
+      "email": "research@netuitive.com"
+    },
+    "distro": {
+      "logo": "http://assets.app.netuitive.com/pkg/logo/netuitive/banner_logo.png",
+      "download": "https://github.com/netuitive-community-packages/netuitive-packages-aws-efs/archive/master.zip"
+    },
+    "compatibility": [
+      "INFRASTRUCTURE:ElasticFileSystem"
```
Comment (TheConnMan) [inline package.json]:
```suggestion
      "AWS:EFS"
```

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+language: node_js
+node_js:
+- '8'
+install:
+- npm i -g metricly-cli
+script:
+- metricly package validate && metricly package lint
+notifications:
+  slack:
+    secure: pacYeCGtuCYKpmiZ/y1v2Y36sEMnOkZcKHCGMMXz+rvTKFtVgFd33QYP5lBAf7xztdT8Dyb2iexVuf1HL93jiaVfTvNTOx/Mp3yxfM6/y7H3r5GJXZebLHUQA+r+53B1LIUi78Y8JWU3S89QFdrkvqKP38RRqu0FAwg1ZFO9RN3uarHC7tD6HW8ZT4ywHitA3OAENKTHD0CdbTl56Jv7hA7iu1J7HktWTjELZGvsbAY5kslf6/yaGSOFmS6gsWU1Uf4mV5Hw31bYhrt79qYxAbaTvbHhKyd6vmvRwoE54RitRVRh55BIgdpI2th7xG5ixZxeKHPqL8kx8ezwUT/JwT9CsNLE2+OaICJvYQyawpE0pJ3rYPopZDAls9e65AzI87mAYBW6UXruMDx6sVkYC05PctdXhIyPv7K7LdlqiNI+tUS3Df6MA86Fj1durzqOm1qxcUvfwciGx7TAi5LxsCYFHzJe3knbKiWw9EsJ+vqax+qEhZTVFbh8hUS2VhRSBoSzHDvX/xsmVNZeS7SE3tM/iqro0tlatGbrWGZvH/mUbWJMB0ihuFdcnZCqS79ErWYRl3jaC6qTHUMkcdwlcTpcTfW+pqFMaZWFHplnOKsTuE9gPwVyITyYfxYiroKdfCV2JRtZpK+wZEw2sENvHiAoGeEwoNffA9DzJVuRaJI=
```
Comment (TheConnMan) [inline .travis.yml]: Is this copied from another package? If so, we'll need to update its value.

**Review by TheConnMan — CHANGES_REQUESTED** [review]
A great start, just a few changes to make.

File: policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        "level": 0,
+        "metric": "aws.efs.burstcreditbalance",
+        "operator": "="
+      }
+    ],
+    "deleted": false,
+    "description": "File system has almost reached its limit",
+    "duration": 900,
+    "enabled": true,
+    "name": "IO Percentage Critical",
```
Comment (rclagett) [inline policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json]: Name should be `AWS EFS - IO Percentage Critical` to follow convention. You might consider something like `AWS EFS - Depleted Burst Credit Balance`.

File: policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    "conditions": [
+      {
+        "analytic": "actual",
+        "level": 0,
+        "metric": "aws.efs.burstcreditbalance",
+        "operator": "="
+      }
+    ],
+    "deleted": false,
+    "description": "File system has almost reached its limit",
```
Comment (rclagett) [inline policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json]: The description should be more detailed.

File: policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    "actions": [
+      {
+        "category": 3,
+        "type": "event"
+      }
+    ],
+    "conditions": [
+      {
+        "analytic": "actual",
+        "level": 0,
```
Comment (rclagett) [inline policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json:12]: Should the level be 0? Perhaps the value should be a bit higher so users know it's very low before it's actually reaches 0. I don't feel strongly about this, just thought I'd pose the question.

File: policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    "conditions": [
+      {
+        "analytic": "actual",
+        "level": 95,
+        "metric": "aws.efs.percentiolimit",
+        "operator": ">"
+      }
+    ],
+    "deleted": false,
+    "description": "Description: There are no burst credits left. The number of burst credits that a file system has is zero.",
```
Comment (rclagett) [inline policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json]: Was this description meant for the other policy?

File: policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json (hunk 7 above)
Comment (rclagett) [reply policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json]: Was this description meant for the other policy?

File: policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+      }
+    ],
+    "deleted": false,
+    "description": "File system has almost reached its limit",
+    "duration": 900,
+    "enabled": true,
+    "name": "IO Percentage Critical",
+    "scope": {
+      "elementTypes": [
+        "SERVER"
```
Comment (rclagett) [inline policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json]: Should be `EFS`

File: policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    "duration": 900,
+    "enabled": true,
+    "name": "Linux ElasticSearch - Cluster Health Degraded to Red",
```
Comment (rclagett) [inline policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json]: This name doesn't match up with what this policy is doing

File: policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    "scope": {
+      "elementTypes": [
+        "SERVER"
```
Comment (rclagett) [inline policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json]: Should be `EFS`

**Review by rclagett — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 5 above)
Comment (ibilous) [reply .travis.yml]: how i can do that ?

File: policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json (hunk 9 above)
Comment (ibilous) [reply policies/aws.efs.percentiolimit.-.elevated.io.consumption.usage.json]: yes :)

File: policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json (hunk 8 above)
Comment (ibilous) [reply policies/aws.efs.burstcreditbalance.-.value.equals.to.zero.json:12]: tbh for me it is not very clear from description ( https://docs.aws.amazon.com/efs/latest/ug/monitoring-cloudwatch.html )  what the  value should be set here. IMO it should be 0.

File: .travis.yml (hunk 5 above)
Comment (TheConnMan) [reply .travis.yml]: Here are the docs: https://docs.travis-ci.com/user/notifications#configuring-slack-notifications

File: analyticConfigurations/computed_metric-efs.json [hunk 13]
```diff
@@ -0,0 +1,10 @@
+{
+  "analyticConfiguration": {
+    "name": "Elastic File System",
+    "scope": {
+      "elementType": "SERVER",
```
Comment (TheConnMan) [inline analyticConfigurations/computed_metric-efs.json]:
```suggestion
      "elementType": "EFS",
```

**Review by TheConnMan — CHANGES_REQUESTED** [review]
This is looking really good. One final change and I think you're good to go!

**Review by TheConnMan — APPROVED** [review]

**Review by yadima — COMMENTED** [review]
All metrics in ACE have baseline set to "false" - is it correct?

**Comment by yadima** [comment]
Widgets are unordered: http://take.ms/lQyiW

**Comment by ibilous** [comment]
> Widgets are unordered: http://take.ms/lQyiW

and how to order them ? should i change somehow  their coordinates ?

**Comment by yadima** [comment]
> > Widgets are unordered: http://take.ms/lQyiW
> 
> and how to order them ? should i change somehow their coordinates ?

It can be done in the "gridstackContents" property.

**Comment by ibilous** [comment]
> > > Widgets are unordered: http://take.ms/lQyiW
> > 
> > 
> > and how to order them ? should i change somehow their coordinates ?
> 
> It can be done in the "gridstackContents" property.

fixed https://i.imgur.com/K46P14L.png

**Comment by ibilous** [comment]
>All metrics in ACE have baseline set to "false" - is it correct?

i think yes

**Comment by ibilous** [comment]
@Grumblesmith @rclagett  could u pls review it once u have some time

**Review by rclagett — APPROVED** [review]

File: analyticConfigurations/computed_metric-efs.json [hunk 14]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (Grumblesmith) [inline analyticConfigurations/computed_metric-efs.json]: There's no need for this file if there aren't any computed metrics

**Review by Grumblesmith — CHANGES_REQUESTED** [review]

File: analyticConfigurations/computed_metric-efs.json (hunk 14 above)
Comment (ibilous) [reply analyticConfigurations/computed_metric-efs.json]: removed

**Pull request merged**